src/arrayMethodSort.js

```diff
@@ -4,8 +4,41 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    let defaultCompareFunction;
+
+    if (typeof compareFunction !== 'function') {
+      defaultCompareFunction = function (a, b) {
+        const stringA = String(a);
+        const stringB = String(b);
+
+        if (stringA < stringB) {
+          return -1;
+        }
+
+        if (stringA > stringB) {
+          return 1;
+        }
+
+        return 0;
+      };
+    } else {
+      defaultCompareFunction = compareFunction;
+    }
```

Comment:
instead of all these checks just specified the default value for the compareFunction